nion/swift/ExportDialog.py

```diff
@@ -34,167 +36,191 @@
-        column.add_stretch()
-
-        def choose() -> None:
-            existing_directory, directory = self.ui.get_existing_directory_dialog(_("Choose Export Directory"),
-                                                                                  self.directory)
-            if existing_directory:
-                self.directory = existing_directory
-                directory_label.text = self.directory
-                self.ui.set_persistent_string("export_directory", self.directory)
-
-        choose_directory_button.on_clicked = choose
-
-        def writer_changed(writer: ImportExportManager.ImportExportHandler) -> None:
-            self.ui.set_persistent_string("export_io_handler_id", writer.io_handler_id)
+        writer = ImportExportManager.ImportExportManager().get_writer_by_id(io_handler_id)
+        if writer:
             self.writer = writer
 
-        file_types_combo_box.on_current_item_changed = writer_changed
+        self.viewmodel = ExportDialogViewModel(True, True, True, True, self.writer, "", self.directory)
+        u = Declarative.DeclarativeUI()
+        self._build_ui(u)
 
-        self.content.add(column)
+        dialog = typing.cast(Dialog.ActionDialog,
+                             Declarative.construct(document_controller.ui, document_controller, u.create_modeless_dialog(self.ui_view, title=_("Export")), self))
+        dialog.add_button(_("Cancel"), self.cancel)
+        dialog.add_button(_("Export"), functools.partial(self.export_clicked, display_items, self.viewmodel))
+        dialog.show()
 
-    def do_export(self, display_items: typing.Sequence[DisplayItem.DisplayItem]) -> None:
-        directory = self.directory
-        writer = self.writer
-        if directory and writer:
+    def choose_directory(self, widget: Declarative.UIWidget) -> None:
+        existing_directory, directory = self.ui.get_existing_directory_dialog(_("Choose Export Directory"),
+                                                                              self.directory)
+        if existing_directory:
+            self.directory = existing_directory
+            self.viewmodel.directory.value = self.directory
+            self.ui.set_persistent_string("export_directory", self.directory)
+
+    def on_writer_changed(self, widget: Declarative.UIWidget, current_index: int) -> None:
+        writer = self.writers[current_index]
+        self.viewmodel.writer.value = writer
+
+    def _build_ui(self, u: Declarative.DeclarativeUI) -> None:
+        self.file_type_index = 0
+        self.writers = ImportExportManager.ImportExportManager().get_writers()
+        writers_names = [getattr(writer, "name") for writer in self.writers]
+        self.file_types = writers_names
+
+        #  self.viewmodel = viewmodel
+
+        # Export Folder
+        directory_label = u.create_row(u.create_label(text="Location:"))
+        directory_text = u.create_row(u.create_label(text=f'@binding(viewmodel.directory.value)'))
```

Comment:
Can you use " instead of ' here? It's a minor thing, but I'd like to be consistent.